web/app/widgets/button/button.widget.js

```diff
@@ -57,17 +57,45 @@
             }
         }
 
+        function onNavigate() {
+            if (!vm.widget.navigate_url)
+                return;
+
+            if (vm.widget.navigate_type === 'dashboard') {
+                $location.url('/view/' + vm.widget.navigate_url);
```

Comment:
To ensure backwards compatibility the model property must remain `navigate_dashboard` - besides it's the dashboard id, not really an url.

In fact, you could get rid of `navigate_type` altogether in the model (not the form object in the settings dialog's scope) and simply derive it by looking at which property is defined:
```
if (vm.widget.navigate_dashboard) {
     // navigate to the dashboard in vm.widget.navigate_dashboard
} else {
     // navigate to the url in vm.widget.navigate_url
}
```
When submitting the settings, the property that isn't used (`navigate_dashboard` or `navigate_url` + `navigate_target`) should be deleted from the model (see below).